server/src/main/java/org/opensearch/action/admin/indices/stats/CommonStatsFlags.java

```diff
@@ -124,6 +128,9 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeEnumSet(includeCaches);
             out.writeStringArrayNullable(levels);
         }
+        if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
Since nodes with older versions will never enter this code path, we will send them the default RestNodesStatsRequest without any additional boolean. This will therefore not break during version upgrade.